src/execution/window_executor.cpp

```diff
@@ -824,76 +912,57 @@ void ExclusionFilter::ResetMask(idx_t row_idx, idx_t offset) {
 	}
 }
 
-//===--------------------------------------------------------------------===//
-// WindowExecutor
-//===--------------------------------------------------------------------===//
-static void PrepareInputExpressions(const vector<unique_ptr<Expression>> &exprs, ExpressionExecutor &executor,
-                                    DataChunk &chunk) {
-	if (exprs.empty()) {
-		return;
+column_t WindowSharedExpressions::RegisterExpr(const unique_ptr<Expression> &expr, Expressions &shared) {
+	auto pexpr = expr.get();
+	if (!pexpr) {
+		return DConstants::INVALID_INDEX;
 	}
 
-	vector<LogicalType> types;
-	for (idx_t expr_idx = 0; expr_idx < exprs.size(); ++expr_idx) {
-		types.push_back(exprs[expr_idx]->return_type);
-		executor.AddExpression(*exprs[expr_idx]);
+	column_t result = shared.size();
+	for (column_t i = 0; i < shared.size(); ++i) {
```

Comment:
Perhaps we can use an [expression_map](https://github.com/duckdb/duckdb/blob/0446ab42e96b6269e78f55293f4096fa10224837/src/include/duckdb/parser/expression_map.hpp#L35) here?